test/npm-utils-spec.js

```diff
@@ -25,33 +25,32 @@
 var npmUtil = require('../src/npm-utils');
 
 describe('npm-utils', function() {
-  it('should get the proxy', function(done) {
-    var promise = npmUtil.proxy();
+  it('should get list of releases', function(done) {
+    var promise = npmUtil.releases('bower', '1.7.7');
     expect(promise).toBeDefined();
 
     promise
-      .then(function(config) {
-        var keys = Object.keys(config);
-        keys.sort();
-        expect(keys).toEqual(['https-proxy', 'proxy']);
+      .then(function(releases) {
+        expect(releases.length).toBeGreaterThan(0);
+        expect(releases).toContains('1.7.7');
       })
       .catch(function() {
-        jasmine.fail('Unable to get tarball URL');
+        jasmine.fail('Unable to get list of releases');
       })
       .finally(function() {
         done();
       });
   });
 
-  it('should get tarball url', function(done) {
-    var promise = npmUtil.tarball('bower', '1.7.7');
+  it('should get tarball', function(done) {
+    var promise = npmUtil.downloadTarball('bower', '1.7.7', __dirname);
```

Comment:
Can you download the tarball to a temporary directory instead of `__dirname` (directory that will be removed after test)?